src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -276,6 +276,9 @@ protected static T Cast<T>(object value)
             return currentDestination;
         }
 
+        private static CodeGenerationOptions ShouldGenerateMethodBody(CodeGenerationOptions options, IPropertySymbol property) =>
+            options.GenerateMethodBodies ? options.WithGenerateMethodBodies(property.IsIndexer || property.IsWriteOnly || property.IsReadOnly || property.Parameters.Length > 0) : options;
```

Comment:
I am making a conservative choice in assuming that if the user has a readonly or writeonly property that they may want to add some custom logic, so we generate a body.  We obviously could generate a getter-only autoprop, but the user would need to write additional code to initialize it anyway.  Do you disagree? would users want getter-only autoproperties to be generated?